src/odemis/acq/stream/_sync.py

```diff
@@ -2000,3 +2001,212 @@ def _runAcquisition(self, future):
+        self._nfr = 0
+
+        self._current_future = None
+        self._acq_state = None
+
+        self._new_acquisition = False
+
+    def acquire(self):
+        # Make sure every stream is prepared, not really necessary to check _prepared
+        f = self.prepare()
+        f.result()
+
+        # TODO: if already acquiring, queue the Future for later acquisition
+        if self._current_future is not None and not self._current_future.done():
+            raise IOError("Cannot do multiple acquisitions simultaneously")
+
+        if not self._acq_done.is_set():
+            if self._acq_thread and self._acq_thread.isAlive():
+                logging.debug("Waiting for previous acquisition to fully finish")
+                self._acq_thread.join(10)
+                if self._acq_thread.isAlive():
+                    logging.error("Previous acquisition not ending")
+
+        est_start = time.time() + 0.1
+        f = model.ProgressiveFuture(start=est_start,
+                                    end=est_start + self.estimateAcquisitionTime())
+        self._current_future = f
+        self._acq_state = RUNNING  # TODO: move to per acquisition
+        self._prog_sum = 0
+        f.task_canceller = self._cancelAcquisition
+
+        # run task in separate thread
+        executeAsyncTask(f, self._runAcquisition, args=(f,))
+        return f
+
+    def _prepareHardware(self):
+        '''
+        Prepare hardware for acquisition and return the best pixel dwelltime value
+        '''
+        # configure the symphotime controller with parameters
+        if hasattr(self._tc_scanner, 'filename') and self._filename is not None:
+            self._tc_scanner.filename.value = self._filename
+        if hasattr(self._tc_scanner, 'directory') and self._directory is not None:
+            self._tc_scanner.directory.value = self._directory
+
+        # The dwell time from the Nikon C2 will set based on what the device is capable of
+        # As a result, we need to recalculate our total dwell time based around this value
+        # and the number of frames we can compute
+        px_dt = min(self._dwellTime.value, self._scanner.dwellTime.range[1])
+        self._nfr = int(math.ceil(self._dwellTime.value / px_dt))  # number of frames
+        px_dt = self._dwellTime.value / self._nfr  # the new dwell time per frame
+        self._scanner.dwellTime.value = px_dt  # try set the C2 dwell time value again.
+        logging.info("Total dwell time: %f s, Pixel Dwell time: %f, collecting %d frames...",
+                     self._dwellTime.value, px_dt, self._nfr)
+
+        self._tc_scanner.dwellTime.value = self._dwellTime.value
+
+        return px_dt
+
+    def _setLightEmission(self, value):
+        # set all light emissions at once to a value
+        em = self._light.emissions.value
+        em = [value] * len(em)
+        self._light.emissions.value = em
+
+    def _runAcquisition(self, future=None):
+
```

Comment:
Please add here a long comment to explain how the acquisition works. (In particular, explain all this game with the time correlator, the scanner and the tc_scanner and why we don't care about the data returned by the time correlator).